controllers/external/util.go

```diff
@@ -107,6 +119,15 @@ type GenerateTemplateInput struct {
 	// +required
 	Template *unstructured.Unstructured
 
+	// TemplateRef is a reference to the template that needs to be cloned.
+	// +required
+	TemplateRef *corev1.ObjectReference
+
+	// IsInfraTemplate is true if the template is infrastructure template reference. If this is true, Template ref
+	// will be added to the cloned object as annotation.
+	// +required
+	IsInfraTemplate *bool
```

Comment:
We should add the cloned-from reference regardless of the template generated, it seems harmless enough

Reply:
When both bootstrap config ref and infrastructure ref are cloned, annotation may be overridden (e.g., machineset_controller).

Reply:
Are we applying the annotation on the cloned object? If so, we should be ok given that there can only be one source for each cloned object

Reply:
You're right! Fixed.